modules/ROOT/pages/6.4-release-notes.adoc

```diff
@@ -0,0 +1,1154 @@
+// TINY-9462
+
+In previous versions of {productname}, elements within the editor.dom could be _modified_ regardless if they were assigned to be `contentEditable = true;` or `contentEditable = false;`.
+
+As a consequence all elements within the root `body` set to `contentEditable = false;` value, were affected by the issue.
+
+The new additions and improvements with the introduction of {productname} 6.4, include a new `isEditable` API been added to the `editor.dom` which now blocks the `editor.insertContent` API and `mceInsertContent` command when the contents return a `true` value if the editor detects the element is a `contentEditable = false;` in a `contentEditable = false;` root.
+
+As a result, any elements that return a `true` value when a specified element is `contentEditable = true;`, will call the new `isEditable` API to action the new blocker to ensure `contentEditable = false;` elements cannot be modified when they are within a root that is set to `contentEditable = false`.
+
+=== New setText and setIcon methods added to menu button and toolbar button apis
+// TINY-9268
+
+In previous versions of {productname}, it was not possible to change the `text` or `icon` of a toolbar button after it had been set for the first time. As a consequence, the menu and toolbar button API, would not extend or update the parent `icon` or `text` directly, based upon what the user had selected within the menubar or toolbar.
+
+In {productname} 6.4, two new additions were added to the menu button API.
+
+. Added `setText` and `setIcon` properties to both the toolbar button `api`, and menu button `api`.
+. Provided the menu button `api` property as an *argument* which will fetch a function that will allow menu items to change the `icon` or `text` of the parent menu button.
+
+As a result, in {productname} 6.4, menu items can change based on the parent menu button `api` properties.
+
+=== New `highlight_on_focus` option which enables highlighting the content area on focus
+// TINY-9277
+
+In previous versions of {productname}, when using developer tools such as Chrome DevTools, the editor content area would not render the `focus` border when the editor was `infocus`.
+
+As a consequence, this issue, whilst creating no impending impacts on the functionality of the editor, would create an inconsistency with the visual UX for the `devTools` environment. As `devTools` is used by developers to inspect the rendered HTML, and troubleshoot issues, the importance for rendering {productname} elements is vital. 
+
+To fix this issue, {productname} 6.4, made some changes to the `focusController` to ensure that when the editor content is `infocus` the editor will render the highlighted border within `devTools`, and when the editor is not `infocus` to remove `focus`.
+
+In {productname} 6.4, the editor now renders the highlighted border when in `focus` which is visible only via `devTools`.
```

Comment:
```suggestion
In previous versions of {productname}, when using developer tools such as Chrome DevTools, the highlighted border around the editor content window should render with the container when is `infocus`.

As a consequence, this issue, whilst creating no impending impacts on the functionality of the editor, would create an inconsistency with the visual UX for the `devTools` environment. As `devTools` is used by developers to inspect the rendered HTML, and troubleshoot issues, the importance for rendering {productname} elements is vital. 

To fix this issue, {productname} 6.4, made some changes to the `focusController` to ensure that when the editor content is `infocus` the editor will render the highlighted border for the editor content window if the below option is integrated into the `tinymce.init()` such as the below:

.Option for setting `highlight_on_focus: true`
[source, js]
----
tinymce.init({
    selector: "textarea",
    plugins: [ 
    ],
    highlight_on_focus: true // This option is required to be set to `true`.
});
----

In {productname} 6.4, when the `highlight_oon_focus option is set to `true`, the editor now renders the highlighted border to the editor content window, when the `focus` is directed to the "textarea" container.
```

Reply:
@shanmen-tiny - TY for the above, please review the amended content, when possible.

Reply:
LGTM!